src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperation.cs

```diff
@@ -57,11 +78,30 @@ public async Task<ExecutionResult<EgressResult>> ExecuteAsync(IServiceProvider s
             }, logger, token);
         }
 
+        public EgressProcessInfo GetEgressProcessInfo()
+        {
+            return _egressProcessInfo;
+        }
+
         public void Validate(IServiceProvider serviceProvider)
         {
             serviceProvider
                 .GetRequiredService<IEgressService>()
                 .ValidateProvider(_egressProvider);
         }
     }
+
+    internal readonly struct EgressProcessInfo
```

Comment:
Let's actually convert this to a class, since we return this value and sometimes it's null.